conans/model/build_info.py

```diff
@@ -159,6 +185,24 @@ def _get_cpp_info():
 
         return self.configs.setdefault(config, _get_cpp_info())
 
+    def raise_if_components_and_non_default_values(self):
+        if (self.includedirs != [DEFAULT_INCLUDE] or
+                self.libdirs != [DEFAULT_LIB] or
+                self.bindirs != [DEFAULT_BIN] or
+                self.resdirs != [DEFAULT_RES] or
+                self.frameworkdirs != [DEFAULT_FRAMEWORK] or
+                self.libs or
+                self.system_libs or
+                self.frameworks or
+                self.defines or
+                self.cflags or
+                self.cxxflags or
+                self.sharedlinkflags or
+                self.exelinkflags or
+                self.build_modules) and self.components:
```

Comment:
Here we should look to the _configs_  too. I would like it to raise if any config element is specified together with the components... 

```python
                def package_info(self):
                    self.cpp_info.release.libs = ["mylibrary"]
                    self.cpp_info.components["int1"].libs.append("libint1")
```